normandy/base/tests/__init__.py

```diff
@@ -29,9 +29,13 @@ def contains(cls, *values):
     @classmethod
     def iso8601(cls):
         def is_iso8601_date(s):
-            # Will throw is it does not match
-            datetime.strptime(s, "%Y-%m-%dT%H:%M:%S.%fZ")
-            return True
+            if not isinstance(s, str):
+                return False
+            try:
+                datetime.strptime(s, "%Y-%m-%dT%H:%M:%S.%fZ")
+                return True
+            except ValueError:
+                return False
```

Comment:
This change makes the errors that come out of this method easier to understand. It gives back something like `"foo" != <Whatever named "datetime">` instead of throwing a `ValueError`.